recipe/meta.yaml

```diff
@@ -1,34 +1,79 @@
+# Notes: this receipt is based on torchvision-feedstock: https://github.com/conda-forge/torchvision-feedstock/blob/master/recipe/meta.yaml
+
 {% set name = "torchani" %}
-{% set version = "2.2" %}
+{% set version = "2.2.1" %}
+{% set torchani_proc_type = "cuda" if cuda_compiler_version != "None" else "cpu" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: c086fe6cabe49a0408d1f1b12aaf14e33b9e6ef0ffb15f3020792934e7c9fbbe
-  
+  git_url: https://github.com/aiqm/{{ name }}.git
+  git_rev: {{ version }}
+
 build:
-  noarch: python
   number: 0
-  script: "{{ PYTHON }} -m pip install . -vv"
+  skip: true  # [win]
+  string: py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}_{{ torchani_proc_type }}                                                  # [cuda_compiler_version == "None"]
+  string: py{{ CONDA_PY }}cuda{{ cuda_compiler_version|replace('.', '') }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}_{{ torchani_proc_type }}  # [cuda_compiler_version != "None"]
+  script:
+    # cuda 10.2 does not have cub, which need to be manually downloaded
+    # - sudo yum install wget unzip -y                                                  # [cuda_compiler_version == '10.2']
+    # However as the current master version of cub has conflict with cuda 10.2, cuaev will not support cuda 10.2 for now.
+    - "{{ PYTHON }} -m pip install . -vvv --no-deps"                                    # [cuda_compiler_version in ("None", '10.2')]
+    - "{{ PYTHON }} -m pip install . --global-option='--cuaev-all-sms' -vvv --no-deps"  # [cuda_compiler_version not in ("None", '10.2')]
 
 requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - {{ compiler('cuda') }}                 # [cuda_compiler_version != "None"]
+    - sysroot_linux-64 ==2.17                # [linux64]
```

Comment:
Do you really need this with the new `cos7` requirement?

Reply:
This is indeed correct
https://conda-forge.org/docs/maintainer/knowledge_base.html?highlight=cuda#using-centos-7